aws/table_aws_wellarchitected_lens.go

```diff
@@ -0,0 +1,243 @@
+			maxLimit = limit
+		}
+	}
+
+	input := &wellarchitected.ListLensesInput{
+		MaxResults: maxLimit,
+	}
+
+	if d.EqualsQualString("lens_status") != "" {
+		input.LensStatus = types.LensStatusType(d.EqualsQualString("lens_status"))
+	}
+	if d.EqualsQualString("lens_name") != "" {
+		input.LensName = aws.String(d.EqualsQualString("lens_name"))
+	}
+	if d.EqualsQualString("lens_type") != "" {
+		input.LensType = types.LensType(d.EqualsQualString("lens_type"))
+	}
+
+	paginator := wellarchitected.NewListLensesPaginator(svc, input, func(o *wellarchitected.ListLensesPaginatorOptions) {
+		o.Limit = maxLimit
+		o.StopOnDuplicateToken = true
+	})
+
+	// List call
+	for paginator.HasMorePages() {
+		output, err := paginator.NextPage(ctx)
+		if err != nil {
+			plugin.Logger(ctx).Error("aws_wellarchitected_lens.listWellArchitectedLenses", "api_error", err)
+			return nil, err
+		}
+
+		for _, item := range output.LensSummaries {
+			// As per the doc(https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_LensSummary.html)
+			// The lens alias is same as lens arn if it is a custom lens.
+			// The lens alias value coming as nil in the case of custom lenses, so we are replacing the null value with lens arn.
+			if item.LensAlias == nil || *item.LensAlias == "" {
+				item.LensAlias = item.LensArn
+			}
+
+			d.StreamListItem(ctx, item)
+
+			// Context can be cancelled due to manual cancellation or the limit has been hit
+			if d.RowsRemaining(ctx) == 0 {
+				return nil, nil
+			}
+		}
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+// We can have two type of lenses (1. AWS official lenses, 2. Custom lenses)
+/*
+	AWS official lenses: arn:aws:wellarchitected:us-east-1::lens/serverless
+	Custom lenses: arn:aws:wellarchitected:us-west-2:123456789012:lens/0123456789abcdef01234567890abcdef
+
+	The API GetLens(https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_GetLens.html) can take two parameter.
+	1. LensAlias:
+		For AWS official lenses, this is either the lens alias, such as serverless, or the lens ARN, such as arn:aws:wellarchitected:us-east-1::lens/serverless
+		For custom lenses, this is the lens ARN, such as arn:aws:wellarchitected:us-west-2:123456789012:lens/0123456789abcdef01234567890abcdef
+	2. LensVersion:
+		The lens version to be retrieved.
+
+	For manipulation the 'LensAlias' values for for both type(AWS official/custom) of lenses is bit wired, also the steampipe query throws error(get call returned 18 results - the key column is not globally unique) in the case of AWS official lenses because the lens aliases are same throughout the regions, so this function can not be use in get config.
+
+	This function has been added as a hydrate function to overcome the error "get call returned 18 results - the key column is not globally unique"
+*/
+
+func getWellArchitectedLens(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+
+	lensArn := *h.Item.(types.LensSummary).LensArn
```

Comment:
@ParthaI we should be sticking to the original required parameters as specified by the API - https://docs.aws.amazon.com/wellarchitected/latest/APIReference/API_GetLens.html

Reply:
Updated the code to use the correct parameter as per mentioned in the API.